tokio/src/timer.rs

```diff
@@ -30,45 +30,44 @@
 //! Wait 100ms and print "Hello World!"
 //!
 //! ```
+//! #![feature(async_await)]
+//!
 //! use tokio::prelude::*;
 //! use tokio::timer::Delay;
 //!
 //! use std::time::{Duration, Instant};
 //!
-//! let when = Instant::now() + Duration::from_millis(100);
 //!
-//! tokio::run({
-//!     Delay::new(when)
-//!         .map_err(|e| panic!("timer failed; err={:?}", e))
-//!         .and_then(|_| {
-//!             println!("Hello world!");
-//!             Ok(())
-//!         })
-//! })
+//! #[tokio::main]
+//! async fn main() {
+//!     let when = tokio::clock::now() + Duration::from_millis(100);
```

Comment:
why the change from `Instant::now`?

Reply:
We should move away from `Instant::now()` in general. `clock::now()` can be hooked into from tests.

Reply:
Just to follow up then we should document that all the timer modules? and maybe even provide api's in delay to auto call out to tokio::clock?